scheduler/scheduler_test.go

```diff
@@ -199,9 +203,27 @@ func TestScheduler(t *testing.T) {
 		e := s.Start()
 		So(e, ShouldBeNil)
 		// create a simple schedule which equals to windowed schedule without start and stop time
-		_, te := s.CreateTask(schedule.NewWindowedSchedule(time.Second, nil, nil, 0), w, false)
+		t, te := s.CreateTask(schedule.NewWindowedSchedule(time.Second, nil, nil, 0), w, false)
 		So(te.Errors(), ShouldBeEmpty)
 
+		Convey("start task", func() {
+			c.timeToWait = 200 * time.Millisecond
```

Comment:
You seem using Go Channel in the implementation. Your tests should use same pattern not timeTowait.